src/main/scala/strawman/collection/Seq.scala

```diff
@@ -26,6 +31,9 @@ trait SeqOps[+A, +CC[X], +C] extends Any
   with ArrayLike[A]
   with Equals {
 
+  protected[this] def c: C
+  protected[this] def seq: Seq[A]
```

Comment:
> I didn't check all the `c` uses, but the `seq` uses don't obviously need to be a seq as far as I can tell

They are both needed to implement `permutations`. We do use `sortBy` which is only available on `Seq` collections.

Note that we can replace `c` with `fromSpecificIterable(coll)` and `seq` with `fromIterable(coll)` (if we put an upper bound on `CC`) but that would be ridiculously inefficient for `ArrayOps` and `StringOps`.

> Anyway, if we really do need this functionality, we should make it public as toSeq. There is no case I can think of where we need it to implement methods, but we can be sure that nobody else needs it.

I don’t think it is useful to have a `toSeq` method on `Seq`. The problem here is a self-type problem for our internal implementation. I’m not sure users are affected by that.